nuhal_tiva/src/pin_tiva.c

```diff
@@ -166,7 +201,23 @@ static void pin_configure(uint32_t pin, enum pin_type type)
     case PIN_PWM:
         GPIOPinConfigure(pin);
         GPIOPinTypePWM(base, mask);
-        break;  
+        break;
+    case PIN_ANALOG:
+        GPIOPinTypeADC(base, mask);
+        tiva_peripheral_enable(SYSCTL_PERIPH_ADC0);
+
+        // Configure Sample Sequencer 3 to use AIN0
+        const uint8_t ain = get_ain_channel(base, mask); // map pin to corresponding AIN channel
+        if (ain == 255) {
+            error(FILE_LINE, "Invalid ADC pin");
+        }
+
```

Comment:
This seems like the wrong place to setup ADC settings. All this should do is set the pin to a certain type, whereas the actual settings for the adc should be separate.